AngularJS/app.js

```diff
@@ -0,0 +1,214 @@
+angular.module("reportApp", ['main']);
+var app = angular.module("main", []);
+
+angular.module('reportApp')
+    .service('reportService', ReportService)
+    .component('reportList', {
+        templateUrl: 'reportList.html',
+        controller: ReportListController
+    })
+
+function ReportListController(reportService) {
+    //Lists all reports in tabular form, with edit and remove options against each report item.
+    var $ctrl = this;
+    var report_config = {
```

Comment:
This config should be part of service as this can be used in multiple components.